### ISSUE TITLE
Add depreciation amount column to depreciation report

### ISSUE THREAD

**Issue by patrict** [snipe/snipe-it#6086]
- Snipe-IT Version: 4.6.2
 - OS: Windows Server 2012
 - Web Server: Apache 2.4.17 via WAMP
 - PHP Version: 5.6.16 via WAMP

Our IT department uses Snipe-IT to manage assets in the company which works great. I would like my finance department to start making use of Snipe-IT as well so that we have a single source of information, as they are currently making use of multiple Excel spreadsheets to manage assets for depreciation and insurance.

In order for my finance department to be productive and not have to break out to Excel, they have requested an additional column on the depreciation report which shows the monthly depreciation amount (Asset cost / depreciation months).

I have already made the changes directly to my local instance of Snipe-IT and everything appears to be working perfectly, however I would obviously not wish to maintain this change through updates. 

This is my first attempt at contributing, and I have read that etiquette dictates that I should open an issue to discuss this before simply creating a pull request. 

Please advise if I can go ahead and create a pull request for the changes described, and also if you have any pointers/advice on what you would like me to include in my changes.

Current changes are very basic:
resources/lang/en/admin/hardware/table.php: Added column heading language string
resources/views/reports/depreciation.blade.php: Added heading and column based on whether depreciation is set for the asset, also checking for divide by zero.

**Comment by Jarli01** [comment]
Just a user asking Isn't the "Diff" field the depreciation to date?

**Comment by patrict** [comment]
Yes it is the depreciation to date. The additional field I am proposing is the monthly depreciation amount - so the calculation is (Asset Cost / Number of Depreciation Months)
Eg: 10,000 / 36 = 277.78

**Comment by Jarli01** [comment]
I guess I could see some value in knowing what the depreciation per that period, but I'm also not in finance so I don't know how useful it would actually be.

**Comment by Jarli01** [comment]
It is also extremely easy to add this formula to the exported document if you chose to do so and determine the depreciation rate. 

Or just on the fly when you look at the report, purchase price/depreciation period = depreciation per month

**Comment by patrict** [comment]
That's correct. My goal is to avoid the need to export and add formula's. 

As mentioned I've already made the changes on my local instance, it was super simple to do, only updated the view and added to the language file for the heading. 

Screenshot attached for reference.

![sc1](https://user-images.githubusercontent.com/8134591/44209731-9990b800-a164-11e8-9298-7a25f098424b.PNG)

**Comment by Jarli01** [comment]
Can you provide your code or make a pull request? My Accounting department would love to have this functionality in Snipe-IT!

**Comment by spiek90** [comment]
Sorry to re-open this issue. I noticed that the field EOL is being used to calculate what the monthly payment should be. I absolutely love this feature! However, shouldn't the field "months" from the depreciation being used to calculate the monthly depreciation? 

In resources/views/reports/depreciation.blade.php, the line 141 can be changed to:
`{{ \App\Helpers\Helper::formatCurrencyOutput(($asset->model->depreciation->months > 0 ? ($asset->purchase_cost / $asset->model->depreciation->months) : 0)) }}`

If the months field is used from the depreciation which is attached to the asset model, you have the option to use a different EOL if desired. 

But perhaps I'm in the wrong here. I've just installed Snipe-IT a few hours ago to be honest :-)

**Comment by patrict** [comment]
Hi @spiek90,

I had assumed that the EOL was automatically populated from the selected depreciation when you set up a new asset model, but could be overridden on the asset model level, which is why I used the EOL value in my monthly depreciation calculation.

However, upon testing now, I had to enter an EOL value manually, even after attaching a depreciation option which had a term value configured.

@snipe am I also misunderstanding? What is the intended purpose of the depreciation entry in relation to the asset model, and should the asset model be getting its EOL value from the depreciation term?

**Comment by henkbotma** [comment]
Hi Patrick. Will you kindly share the code for the monthly depreciation?
